### ISSUE TITLE
Readonly mode

### ISSUE THREAD

**Issue by gafderks** [mbraak/django-mptt-admin#425]
If a logged in user does not have change permission, a `PermissionDenied` is raised in `changelist_view`. Is it possible to allow accessing the changelist view with the editing functionality disabled?

I found #347 (`is_drag_and_drop_enabled`), but this still raises the `PermissionDenied` if the user does not have change permission.

**Comment by mbraak** [comment]
I'll have a look

**Comment by mbraak** [comment]
I'm working on a pr: #426

**Comment by mbraak** [comment]
This is what the pr does. Would this work in your situation?

* 'Add' link: only display it if the admin user has the 'Add' permission
* 'Edit' link: only display it if the admin user has the 'Edit' permission
  * Otherwise: still display it but change the text to 'view'
* Drag and drop: only enable it if the admin user has the 'Edit' permission
  * It's also possible to disable drag and drop by overriding the 'is_drag_and_drop_enabled' method.

**Comment by gafderks** [comment]
Yes, that would be perfect!

**Comment by mbraak** [comment]
This is included in the new 2.4.0 version.

**Comment by gafderks** [comment]
I'm getting the following error, unfortunately:
```
  File "/home/runner/.local/share/virtualenvs/dbase-yH0w66Ia/lib/python3.10/site-packages/django_mptt_admin/django_mptt_admin_mixin.py", line 104, in changelist_view
    tree_mouse_delay=self.get_tree_mouse_delay(),
AttributeError: 'CategoryAdmin' object has no attribute 'get_tree_mouse_delay'```

Do you have any clue what could be the reason for this?

**Comment by mbraak** [comment]
I'll have a look.

**Comment by mbraak** [comment]
This is fixed in version 2.4.1

**Comment by gafderks** [comment]
Thank you! The new version is working well!
